packages/flutter/lib/src/widgets/container.dart

```diff
@@ -184,6 +184,10 @@ class DecoratedBox extends SingleChildRenderObjectWidget {
 /// `width`, `height`, and [constraints] arguments to the constructor override
 /// this.
 ///
+/// A container by itself is hittable during a hit test if and only if it has
+/// [color], [decoration], or [foregroundDecoration]. By using decorations,
+/// the hittability will be decided by [Decoration.hitTest].
```

Comment:
I think this is true, but it's a little bit difficult to parse. Maybe

> By default, containers return false for all hit tests. If the [color] property is specified, the hit testing is handled by [ColoredBox], which always returns true. If the [decoration] or [foregroundDecoration] properties are specified, hit testing is handled by [Decoration.hitTest], which also always returns true.

It's a little unsettling to be documenting such behavior of other widgets, but I don't see how that could change without some kind of break anyway at this point.

Reply:
I don't think it's fair to say `Decoration.hitTest` always returns true, since the subclass will very likely override it, such as in https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/painting/shape_decoration.dart#L286.